core/src/main/java/io/grpc/internal/ServerCallImpl.java

```diff
@@ -169,6 +169,7 @@ private void sendMessageInternal(RespT message) {
         stream.flush();
       }
     } catch (RuntimeException e) {
+      log.warning("Server sendMessage() failed with Error: " + e);
```

Comment:
Don't use string concatenation for logging exceptions. It loses the cause information. Use the `Level, String, Throwable` overload. You also say "Error" here which is incorrect.

In this case, the bug is we should have called `internalClose()` instead of `close()`. The logging in `internalClose()` is using Status instead of Exception, which would make it hard to know which user code was busted. I think we should change `internalClose(Status)` to `internalClose(StatusRuntimeException)` or `internalClose(Throwable)`.